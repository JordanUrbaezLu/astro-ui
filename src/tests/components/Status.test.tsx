import { screen, render } from "@testing-library/react";
import {
  Status,
  StatusVariant,
} from "../../components/Status/Status";
import { axe } from "jest-axe";

describe("Status", () => {
  test("Should render component correctly.", () => {
    const { container } = render(<Status />);
    expect(container.firstChild).toMatchSnapshot();
  });
});

describe("Accessibility", () => {
  test("Should have no accessibility violations.", async () => {
    const { container } = render(<Status />);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});

describe("Props", () => {
  test.each<StatusVariant>(["online", "offline", "away", "busy"])(
    "Should render variants correctly.",
    (variant) => {
      render(<Status variant={variant} label={variant} />);
      expect(screen.getByRole("status").childNodes[0]).toHaveClass(
        variant
      );
    }
  );
});
