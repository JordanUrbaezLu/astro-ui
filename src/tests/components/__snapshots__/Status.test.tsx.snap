// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Status Should render component correctly. 1`] = `
<div
  class="statusContainer"
  role="status"
  title="Offline"
>
  <span
    class="status offline"
  />
</div>
`;
